### DISCUSSION TITLE
Can the 485 chip flow control function be added?

### DISCUSSION THREAD

**Discussion by ShiPC123** [stm32duino/Arduino_Core_STM32#2020] [Q&A]
In industrial control, it is often necessary to use the 485,485 chip for half duplex control, receiving and sending depends on the 485 DE pin, high power usually for sending mode, low power usually for receiving mode. Can you add the following functions to the code: When the TXD is sending data over a serial port, one I/O of the MCU changes to the high level and automatically changes to the low level after the TXD is sent

**Comment by fpistm** [comment]
Hi @ShiPC123 
could you be more precise on this request? which chip? ....

**Reply by ShiPC123** [reply]
such as SP3485 ,What I need is 485 automatic flow control, on the stm32cubeMX configuration interface is as follows: http://img.icspec.com/article/3e3856e0-b603-11ec-8c94-b8ca3a6cb5c4.png

**Reply by fpistm** [reply]
OK understood anyway, I will not have time to support this. Feel free to contribute if you are able to implement it.

**Comment by nanoparticle** [comment]
Hello, I am in the process of designing a board that will use an STM32 MCU with an RS485 transceiver, is there any update with regards to this question? If not, I would be happy to help implement the feature.

**Reply by ShiPC123** [reply]
hello,I control the direction of the 485 chip by calculating the time required to transmit a byte under different baud rates and using the method of delay

**Reply by nanoparticle** [reply]
Thanks for the reply. I would really prefer to have a more robust solution that uses the hardware flow control peripheral, because my use-case is a high performance real-time application. After the board is designed I'll look into submitting a PR for this feature.
